native/swift/Sources/wordpress-api/WordPressAPI.swift

```diff
@@ -97,6 +107,59 @@ public actor WordPressAPI {
         self.requestBuilder.siteSettings()
     }
 
+#if PROGRESS_REPORTING_ENABLED
+    public func uploadMedia(
+        params: MediaCreateParams,
+        fromLocalFileURL localFileURL: URL,
+        fulfilling progress: Progress,
+        mimeType: String? = nil,
+    ) async throws -> MediaRequestCreateResponse {
+        precondition(localFileURL.isFileURL)
+        precondition(progress.completedUnitCount == 0 && progress.totalUnitCount > 0)
+
+        let requestId = WpUuid()
+
+        let fileContentType: String
+        if let mimeType {
+            fileContentType = mimeType
+        } else if let mimeType = UTType(filenameExtension: localFileURL.pathExtension)?.preferredMIMEType {
+            fileContentType = mimeType
+        } else {
+            fileContentType = "application/octet-stream"
+        }
+
+        let cancellable = requestExecutor.progress(forRequestWithId: requestId.uuidString())
+            .flatMap { $0.publisher(for: \.fractionCompleted, options: .new) }
+            // The apps expect progress to be updated on the main thread. We probably should look into
+            // refactoring this part (starting from the app side).
+            .receive(on: DispatchQueue.main)
+            .sink { [weak progress] fraction in
+                guard let progress else { return }
+                progress.completedUnitCount = Int64(fraction * Double(progress.totalUnitCount))
```

Comment:
In my testing `totalUnitCount` is always zero, so I'm not sure what this block is doing?